File: src/main/java/pl/arturczopek/mycoach/model/add/NewTraining.java
package pl.arturczopek.mycoach.model.add;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.sql.Date;
import java.util.List;

/**
 * @Author Artur Czopek
 * @Date 14-04-2017
 */

@Data
public class NewTraining {

    @JsonProperty(required = true)
    private long setId;

    @JsonProperty(required = true)
    private List<NewExerciseSession> exerciseSessions;

    @JsonProperty(required = true)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy")
    private Date trainingDate;
}

File: src/main/kotlin/pl/arturczopek/mycoach/model/add/NewTraining.kt
package pl.arturczopek.mycoach.model.add;

import com.fasterxml.jackson.annotation.JsonFormat
import java.sql.Date

/**
 * @Author Artur Czopek
 * @Date 14-04-2017
 */

data class NewTraining(
        val setId: Long,
        val exerciseSessions: List<NewExerciseSession>,
        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd") val trainingDate: Date
)
